src/compiler/checker.ts

```diff
@@ -2807,15 +2807,19 @@ module ts {
             }
         }
 
-        function getUnwidenedFlagOfTypes(types: Type[]): TypeFlags {
-            return forEach(types, t => t.flags & TypeFlags.Unwidened) || 0;
+        function getWideningFlagsOfTypes(types: Type[]): TypeFlags {

```

Comment:
It is clear what this does, but please put a comment explaining when it should be called. It would seem very easy to forget to call it.